lib/src/prop_validator.dart

```diff
@@ -0,0 +1,21 @@
+/// Typedef for `Component2.propTypes` - used to check the validity of one or more of the [props].
+///
+/// [info] is a [PropValidatorInfo] class that contains metadata about the prop referenced as
+/// the key within the `Component2.propTypes` map.
+/// `propName`, `componentName`, `location` and `propFullName` are available.
+typedef PropValidator<T> = Error Function(T props, PropValidatorInfo info);
```

Comment:
This should be `dynamic` as per the AC, allowing `return null` to be omitted without there being any hints
```suggestion
typedef PropValidator<T> = dynamic Function(T props, PropValidatorInfo info);
```

Reply:
![nope](https://user-images.githubusercontent.com/1750797/68408189-25b04600-0142-11ea-8766-ad1a59e16154.gif)

Reply:
😿

Reply:
its so weird, it like like implicitly types it based on the conditional return... whattt.